### PR TITLE
Allow Multiple ParentDataWidgets to be applied on the same RenderObject

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
Relax an existing assert so this is allowed:

```dart
class _TheatreParentData extends StackParentData { 
  Object? extraInfo;
}

class _TheatreParentDataWidget extends ParentDataWidget<_TheatreParentData> { 
  ...
}

Widget build(context) { 
  return Positioned( 
    child: _TheatreParentDataWidget(
      info: extraInfo,
      child: overlayChildRenderObjectWidget,
    ),
  );

}
```

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/test/widgets/parent_data_test.dart [hunk 1]
```diff
@@ -277,8 +277,8 @@ void main() {
       equalsIgnoringHashCodes(
         'Incorrect use of ParentDataWidget.\n'
         'The following ParentDataWidgets are providing parent data to the same RenderObject:\n'
-        '- Positioned(left: 7.0, top: 6.0) (typically placed directly inside a Stack widget)\n'
-        '- Positioned(top: 5.0, bottom: 8.0) (typically placed directly inside a Stack widget)\n'
+        '- Positioned(left: 7.0, top: 6.0) (for StackParentData, typically placed directly inside a Stack widget)\n'
```
Comment (LongCatIsLooong) [inline packages/flutter/test/widgets/parent_data_test.dart:280]: This is the only test that's verifying old assert as far as I can tell

**Comment by Hixie** [comment]
This seems like it removes a really important guardrail. Can you talk more about why this is ok?

**Comment by goderbauer** [comment]
(triage): @LongCatIsLooong Do you still have plans to come back to this one and answer the question above?

**Comment by LongCatIsLooong** [comment]
yeah I'll try to get back to this one this or the coming week

**Comment by goderbauer** [comment]
(triage): @LongCatIsLooong do you still have plans to come back to this one?

**Comment by LongCatIsLooong** [comment]
well I'll come back in a few weeks I guess.

**Pull request closed without merging**